File: python/45_bicycle_chain.py
n = int(input().strip())
a = list(map(int, input().split()))
m = int(input().strip())
b = list(map(int, input().split()))
best = 0
cnt = 0
for ai in a:
    for bj in b:
        if bj % ai == 0:
            r = bj // ai
            if r > best:
                best = r
                cnt = 1
            elif r == best:
                cnt += 1
print(cnt)

File: cpp/45_bicycle_chain.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    int n, m;
    if (!(cin >> n))
        return 0;
    vector<int> a(n);
    for (int i = 0; i < n; i++)
        cin >> a[i];
    cin >> m;
    vector<int> b(m);
    for (int j = 0; j < m; j++)
        cin >> b[j];
    int best = 0, cnt = 0;
    for (int i = 0; i < n; i++)
        for (int j = 0; j < m; j++)
            if (b[j] % a[i] == 0)
            {
                int r = b[j] / a[i];
                if (r > best)
                {
                    best = r;
                    cnt = 1;
                }
                else if (r == best)
                    cnt++;
            }
    cout << cnt << "\n";
}
